### ISSUE TITLE
add security notice to neoj4 graph in code & docs

### ISSUE THREAD

**Issue by bracesproul** [langchain-ai/langchainjs#2976]
Security notice from python:
```
    *Security note*: Make sure that the database connection uses credentials
        that are narrowly-scoped to only include necessary permissions.
        Failure to do so may result in data corruption or loss, since the calling
        code may attempt commands that would result in deletion, mutation
        of data if appropriately prompted or reading sensitive data if such
        data is present in the database.
        The best way to guard against such negative outcomes is to (as appropriate)
        limit the permissions granted to the credentials used with this tool.
        See https://python.langchain.com/docs/security for more information.
```

Also, find other places notice should be applied

**Comment by jacoblee93** [comment]
CC @tomasonjo since it can technically run arbitrary queries

**Comment by tomasonjo** [comment]
LGTM, you could recommend read only users, etc... also guardrails in prompt

**Comment by jacoblee93** [comment]
Agreed, worth adding

**Comment by bracesproul** [comment]
Closed by #2998
